packages/core/src/transport/batch.spec.ts

```diff
@@ -1,150 +1,95 @@
 
-    batch.add({ message: '60 bytes - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' })
-    expect(sendSpy).toHaveBeenCalledWith({
-      data: '{"message":"50 bytes - xxxxxxxxxxxxxxxxxxxxxxxxx"}',
-      bytesCount: 50,
-      flushReason,
-    })
+    flushController.notifyFlush()
 
-    batch.flush(flushReason)
-    expect(sendSpy).toHaveBeenCalledWith({
-      data: '{"message":"60 bytes - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}',
-      bytesCount: 60,
+    expect(transport.send.calls.mostRecent().args[0]).toEqual({
+      data: '{"message":"hello"}',
+      bytesCount: SMALL_MESSAGE_BYTES_COUNT,
       flushReason,
     })
   })
 
-  it('should consider separators when computing the byte count', () => {
-    batch.add({ message: '30 bytes - xxxxx' }) // batch: 30 sep: 0
-    batch.add({ message: '30 bytes - xxxxx' }) // batch: 60 sep: 1
-    batch.add({ message: '39 bytes - xxxxxxxxxxxxxx' }) // batch: 99 sep: 2
-
-    expect(sendSpy).toHaveBeenCalledWith({
-      data: '{"message":"30 bytes - xxxxx"}\n{"message":"30 bytes - xxxxx"}',
-      bytesCount: 61,
-      flushReason,
-    })
-  })
+  it('should add message to the flush controller', () => {
+    batch.add(SMALL_MESSAGE)
 
-  it('should call send one time when the byte count is too high and the batch is empty', () => {
-    const message = '101 bytes - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
-    batch.add({ message })
-    expect(sendSpy).toHaveBeenCalledWith({ data: `{"message":"${message}"}`, bytesCount: 101, flushReason })
+    expect(flushController.willAddMessage).toHaveBeenCalledOnceWith(SMALL_MESSAGE_BYTES_COUNT)
+    expect(flushController.didAddMessage).toHaveBeenCalledOnceWith()
   })
 
-  it('should flush the batch and send the message when the message is too heavy', () => {
-    const message = '101 bytes - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+  it('should consider separators when adding message', () => {
+    batch.add(SMALL_MESSAGE)
+    batch.add(SMALL_MESSAGE)
+    batch.add(SMALL_MESSAGE)
 
-    batch.add({ message: '50 bytes - xxxxxxxxxxxxxxxxxxxxxxxxx' })
-    batch.add({ message })
-    expect(sendSpy).toHaveBeenCalledTimes(2)
+    expect(flushController.willAddMessage.calls.allArgs()).toEqual([
+      [SMALL_MESSAGE_BYTES_COUNT],
+      [SMALL_MESSAGE_BYTES_COUNT + SEPARATOR_BYTES_COUNT],
+      [SMALL_MESSAGE_BYTES_COUNT + SEPARATOR_BYTES_COUNT],
+    ])
   })
 
-  it('should flush after timeout', () => {
-    const clock = sinon.useFakeTimers()
-    batch = new Batch(transport, BATCH_MESSAGES_LIMIT, BATCH_BYTES_LIMIT, MESSAGE_BYTES_LIMIT, 10, pageExitObservable)
-    batch.add({ message: '50 bytes - xxxxxxxxxxxxxxxxxxxxxxxxx' })
-    clock.tick(100)
+  it('should consider separators when replacing messages', () => {
+    batch.add(SMALL_MESSAGE)
+    batch.upsert(SMALL_MESSAGE, 'a')
 
-    expect(sendSpy).toHaveBeenCalled()
+    flushController.willAddMessage.calls.reset()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why do you need this?

Reply:
I need to call `reset` to reset the number of calls after the test setup, so I can use `expect().toHaveBeenCalledOnceWith(...)` (only the last call is relevent to this test case)